provider/runbook_resource_test.go

```diff
@@ -0,0 +1,325 @@
+		if expected != got {
+			return fmt.Errorf("Unexpected type. Expected: %s, got: %s", expected, got)
+		}
+
+		expected, got = runbookResource.Primary.Attributes["description"], runbookResponse.Description
+		if expected != got {
+			return fmt.Errorf("Unexpected description. Expected: %s, got: %s", expected, got)
+		}
+
+		if len(runbookResponse.Steps) != 1 {
+			return fmt.Errorf("Unexpected number of steps. Expected 1 step, got: %v", len(runbookResponse.Steps))
+		}
+
+		for index, step := range runbookResponse.Steps {
+			key := fmt.Sprintf("steps.%d", index)
+			if runbookResource.Primary.Attributes[key+".name"] != step.Name {
+				return fmt.Errorf("Unexpected runbook step name. Expected %s, got: %s", step.Name, runbookResource.Primary.Attributes[key+".name"])
+			}
+			if runbookResource.Primary.Attributes[key+".action_id"] != step.ActionID {
+				return fmt.Errorf("Unexpected runbook step action_id. Expected %s, got: %s", step.Name, runbookResource.Primary.Attributes[key+".action_id"])
+			}
+			// TODO: Test that config matches
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckRunbookResourceDestroy() resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client, err := firehydrant.NewRestClient(os.Getenv("FIREHYDRANT_API_KEY"))
+		if err != nil {
+			return err
+		}
+
+		for _, stateResource := range s.RootModule().Resources {
+			if stateResource.Type != "firehydrant_runbook" {
+				continue
+			}
+
+			if stateResource.Primary.ID == "" {
+				return fmt.Errorf("No instance ID is set")
+			}
+
+			_, err := client.Runbooks().Get(context.TODO(), stateResource.Primary.ID)
+			if err == nil {
+				return fmt.Errorf("Runbook %s still exists", stateResource.Primary.ID)
+			}
+		}
+
+		return nil
+	}
+}
+
+func testAccRunbookResourceConfig_basic(rName string) string {
+	return fmt.Sprintf(`
+data "firehydrant_runbook_action" "create_incident_channel" {
+  slug             = "create_incident_channel"
+  integration_slug = "slack"
+  type             = "incident"
+}
+
+resource "firehydrant_runbook" "test_runbook" {
+  name        = "test-runbook-%s"
+  type        = "incident"
+
+  steps {
+    name      = "Create Incident Channel"
+    action_id = data.firehydrant_runbook_action.create_incident_channel.id
+    config = {
+      channel_name_format = "-inc-{{ number }}"
+    }
+  }
+}`, rName)
+}
+
+func testAccRunbookResourceConfig_update(rName string) string {
+	return fmt.Sprintf(`
+data "firehydrant_runbook_action" "notify_channel" {
+  slug             = "notify_channel"
+  integration_slug = "slack"
+  type             = "incident"
+}
+
+resource "firehydrant_runbook" "test_runbook" {
+  name        = "test-runbook-%s"
+  type        = "incident"
+  description = "test-description-%s"
+
+  steps {
+    name    = "Notify Channel"
+    action_id = data.firehydrant_runbook_action.notify_channel.id
+    config = {
+      "channels" = "#incidents"
+    }
+  }
+}`, rName, rName)
```

Comment:
Had to look this up because I was like why not have the ability to interpolate the string, but I can see with static typings this make sense for compile time checking https://channaly.medium.com/string-interpolation-in-golang-ecb3bcb2d600.

Reply:
I constantly forget that I can't do string interpolation the same way you can in ruby and other languages until it yells at me 😅